### PR TITLE
CMR-6679 CI build updates

### PR DESCRIPTION
**Pull request by jaybarra**
Fixes to handle the update build in the CI environment.

### CODE REVIEW

File: redis-utils-lib/project.clj [hunk 1]
```diff
@@ -42,16 +44,15 @@
...
-                                     [lambdaisland/kaocha-cloverage "1.0.63"]
+             :kaocha {:dependencies [[lambdaisland/kaocha "1.0.732"]
+                                     [lambdaisland/kaocha-cloverage "1.0.75"]
                                      [lambdaisland/kaocha-junit-xml "0.0.76"]]}}
   :aliases {"bikeshed" ["with-profile" "lint" "bikeshed" "--max-line-length=100"]
             "check-deps" ["with-profile" "lint" "ancient" ":all"]
             "check-sec" ["with-profile" "security" "dependency-check"]
             "eastwood" ["with-profile" "lint" "eastwood" "{:namespaces [:source-paths]}"]
-            "pull-docker-images" ["shell" "docker" "pull" "redis:6"]
+            "pull-docker-images" ["shell" "docker" "pull" ~(str "redis:" redis-version)]
```
Comment (mschmele) [inline redis-utils-lib/project.clj:54]: This feels a bit overkill, and I think the only reason I didn't feel that way with the elasticsearch version is there wasn't the need for a `str` call. Are the redis and elasticsearch versions being called elsewhere? If not I don't really see the benefit of this, even if it is nifty

File: system-int-test/test/cmr/system_int_test/search/collection_concept_retrieval_test.clj [hunk 2]
```diff
@@ -161,23 +165,23 @@
     (testing "retrieval of HTML with extension"
       (let [response (search/retrieve-concept
                        (:concept-id coll1) nil {:query-params {:token user1-token}
-                                                :url-extension "html"})
-            _ (is (= 200 (:status response)))]
+                                                :url-extension "html"})]
+        (is (= 200 (:status response)))
```
Comment (mschmele) [inline system-int-test/test/cmr/system_int_test/search/collection_concept_retrieval_test.clj:169]: 👏 

I hate the `_` bindings in `let` blocks

**Review by mschmele — APPROVED** [review]

File: access-control-app/project.clj [hunk 3]
```diff
@@ -105,9 +105,17 @@
             ;; Kaocha test aliases
             ;; refer to tests.edn for test configuration
             "kaocha" ["with-profile" "+kaocha" "run" "-m" "kaocha.runner"]
-            "itest" ["kaocha" "--focus" "integeration"]
-            "utest" ["kaocha" "--focus" "unit"]
-            "ci-test" ["kaocha" "--profile" ":ci"]
+            "itest" ["kaocha" "--focus" ":integration"]
```
Comment (ygliuvt) [inline access-control-app/project.clj:108]: Add comments on what these targets are for and explain why the ci build is different from local.

File: bin/functions/test.sh [hunk 4]
```diff
@@ -76,6 +84,60 @@ function test_cicd {
     cmr stop uberjar dev-system
 }
 
+function test_ci-utest {
+    cmr build all
+    cmr start uberjar dev-system
```
Comment (ygliuvt) [inline bin/functions/test.sh:89]: Do we need to start dev-system in order to run unit tests?

**Comment by ygliuvt** [comment]
We should add a high level description on Kaocha and how different groups are set up in ci and basic conventions to follow (e.g. test file must end with `test.clj` etc.) in a README somewhere.

**Review by ygliuvt — CHANGES_REQUESTED** [review]
Good job! Need some high level documentation on ci setup. It could be in a README or a link in README to a wiki page.

File: system-int-test/README.md [hunk 5]
```diff
@@ -4,8 +4,45 @@ System integration test for the CMR projects. It will ingest concepts using CMR
 
 ## Usage
 
-lein test
+To run the system integration test suite, ensure all required services of CMR are running.
```
Comment (ygliuvt) [inline system-int-test/README.md:7]: Great documentation.

**Review by ygliuvt — APPROVED** [review]
Bamboo build needs to be modified to produce the necessary artifacts needed by downstream deployment plans (e.g. ES AMI build and ingest-driver, etc.), but that is outside of the scope of this PR.

**Pull request merged**